conn.go

```diff
@@ -53,8 +53,17 @@ type Conn struct {
 	closed bool
 	outLck sync.RWMutex
 
-	signals    []chan<- *Signal
-	signalsLck sync.Mutex
+	signals           []chan<- *Signal
+	signalsLck        sync.Mutex
+	shutdownSigWorker chan int
+
+	pendingSignalChannels    []chan<- *Signal
+	pendingSignalChannelsLck sync.Mutex
+
+	signalQueue       []*Signal
+	signalQueueLck    sync.Mutex
+	signalQueueWakeUp chan int
+	doWakeUp          bool
```

Comment:
Convention is that everything the mutex is actually protecting is grouped right below it, right?